agent/security_helper.py

```diff
@@ -264,3 +267,67 @@ def mysql_root_access():
     except (FileNotFoundError, PermissionError):
         # Can't execute mysql client.
         pass
+
+
+def cpu_vulnerabilities():
+    """
+    Query sysfs for CPU vulnerabilities mitigation.
+    :return: A dict where
+        'vendor': "Vendor ID" field returned by lscpu. Possible values: GenuineIntel, AuthenticAMD, ARM.
+        'vulnerable': False if not vulnerable, True if vulnerable, None if in doubt. Present if vendor is GenuineIntel.
+        'mitigations_disabled': whether any mitigation was disabled in kernel cmdline. Present if vulnerable is None.
+    """
+    from sh import lscpu
+
+    lscpu_stdout = lscpu('-J').stdout
+    lscpu_json = json.loads(lscpu_stdout)
+    vendor_id = next(e['data'] for e in lscpu_json['lscpu'] if e['field'] == "Vendor ID:")
+    res = {'vendor': vendor_id}
+    if vendor_id != "GenuineIntel":
+        # Not an Intel CPU, most probably not vulnerable
+        return res
+
+    sys_vulnerabilities = Path('/sys/devices/system/cpu/vulnerabilities')
+    if not sys_vulnerabilities.is_dir():
+        # Directory does not exist: either smth is bind-mounted over it or the kernel is too old.
+        vulnerable = None
+    else:
+        vulnerable = False
+        for name in ('l1tf', 'mds', 'meltdown', 'spectre_v1', 'spectre_v2', 'spec_store_bypass'):
+            f = sys_vulnerabilities / name
+            if f.exists():
+                # If CPU is not prone to this vulnerability the status file will start with
+                # 'Not affected' or 'Mitigation: ...'. Otherwise it will start with 'Vulnerable: ...'.
+                if f.read_text().startswith('Vulnerable'):
+                    vulnerable = True
+                    break
+            else:
+                # Status file does not exist: smth is bind-mounted over it or the kernel is not completely patched.
+                vulnerable = None
+                break
```

Comment:
Why break? Aren't you going to check other vulns' files?

Reply:
No, one is enough already.